MonstieWash/Assets/Scripts/UIManager.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class UIManager : MonoBehaviour
+{
+    [SerializeField] public TextMeshProUGUI Text;
+    [SerializeField] public Image Clipboard;
+    [SerializeField] public Animator CBAnimator;
+
+    private Dictionary<string, float> m_Tasks;
+    private bool m_UIVisible;
+    // Start is called before the first frame update
+    private void Awake()
+    {
+        m_Tasks = gameObject.GetComponent<TaskTracker>().m_taskProgress;
+        InputManager.Inputs.OnToggleUI += Inputs_OnToggleUI;
```

Comment:
This is a no-no unfortunately. This line should be done in `OnEnable` and the event should be unsubscribed from in `OnDisable`. Other than that, this is good.
```suggestion
        private void OnEnable()
        {
            InputManager.Inputs.OnToggleUI += Inputs_OnToggleUI;
        }
        
        private void OnDisable()
        {
            InputManager.Inputs.OnToggleUI -= Inputs_OnToggleUI;
        }
```